lib/logstash/inputs/LogStashRecordProcessor.rb

```diff
@@ -0,0 +1,65 @@
+# encoding: utf-8
+#
+#Copyright 2015 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+#Licensed under the Apache License, Version 2.0 (the "License");
+#you may not use this file except in compliance with the License.
+#You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+#Unless required by applicable law or agreed to in writing, software
+#distributed under the License is distributed on an "AS IS" BASIS,
+#WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#See the License for the specific language governing permissions and
+#limitations under the License.
+#
+require "java"
+begin
+  require 'jar-dependencies'
+  require_jar( 'com.amazonaws', 'amazon-kinesis-client', '1.6.0' )
+  require_jar( 'log4j', 'log4j', '1.2.17' )
+end
+java_import "com.amazonaws.services.kinesis.clientlibrary.types.ShutdownReason"
+java_import "java.lang.IllegalStateException"
+java_import "org.apache.log4j.LogManager"
+
+class LogStashRecordProcessor
+  include com.amazonaws.services.kinesis.clientlibrary.interfaces::IRecordProcessor
+
+  attr_accessor :queue, :shard_id
+
+  def initialize(queue)
+    # Workaround for IRecordProcessor.initialize(String shardId) interfering with constructor.
+    # No good way to overload methods in JRuby, so deciding which was supposed to be called here.
```

Comment:
so what you aim to do here is to overload the constructor? can you educate me a bit more about your issue? not sure I got it, xD!